### PR TITLE
feat: Improve package publish mechanism

### PR DESCRIPTION
**Pull request by IgnasCi**
https://hostingers.atlassian.net/browse/AUTO-3499

package publishing will include both:

Google cloud artifactory (GCA) / github registry.

Once full migration is completed, github registry publishing will be fully deprecated.

previous mechanism has never been used: 
image

### CODE REVIEW

File: .github/workflows/npm-publish-github-packages.yml [hunk 1]
```diff
@@ -1,26 +1,76 @@
...
-    runs-on: ubuntu-latest
-    permissions:
-      contents: read
-      packages: write
+    runs-on: fireactions-fast-c2-m4
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - name: Checkout repository
+        uses: actions/checkout@v3
```
Comment (MantasGurskis) [inline .github/workflows/npm-publish-github-packages.yml:14]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/npm-publish-github-packages.yml [hunk 2]
```diff
@@ -1,26 +1,76 @@
...
         with:
-          node-version: 16
+          token: ${{ secrets.ORG_GITHUB_TOKEN }}
```
Comment (MantasGurskis) [inline .github/workflows/npm-publish-github-packages.yml:16]: use gh app token

File: .github/workflows/npm-publish-github-packages.yml [hunk 3]
```diff
@@ -1,26 +1,76 @@
...
+          ref: master
+
+      # Setup .npmrc file to publish to GitHub Packages
+      - name: Configure node and Github private registry
+        uses: actions/setup-node@v3
```
Comment (MantasGurskis) [inline .github/workflows/npm-publish-github-packages.yml:21]:
```suggestion
        uses: actions/setup-node@v4
```

File: .github/workflows/npm-publish-github-packages.yml (hunk 2 above)
Comment (MantasGurskis) [reply .github/workflows/npm-publish-github-packages.yml:16]: same in other places

File: .github/workflows/npm-publish-github-packages.yml (hunk 3 above)
Comment (IgnasCi) [reply .github/workflows/npm-publish-github-packages.yml:21]: do not pay attention to github script. It is just a duplicate from other repos. Will deprecate soon after full migration.

File: .github/workflows/npm-publish-github-packages.yml [hunk 4]
```diff
@@ -1,26 +1,76 @@
 # This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
 # For more information see: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
 
-name: Release Package
-
+name: Publish package
 on:
-  release:
-    types: [created]
+  pull_request:
```
Comment (MantasGurskis) [inline .github/workflows/npm-publish-github-packages.yml:6]: are we sure that we want to change the trigger?

**Review by MantasGurskis — APPROVED** [review]

File: .github/workflows/npm-publish-github-packages.yml (hunk 4 above)
Comment (MantasGurskis) [reply .github/workflows/npm-publish-github-packages.yml:6]: but its fine i guess

File: .github/workflows/npm-publish-github-packages.yml [hunk 5]
```diff
@@ -1,26 +1,76 @@
...
+          app-id: ${{ secrets.HOSTINGERBOT_APP_ID }}
+          private-key: ${{ secrets.HOSTINGERBOT_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+          repositories: |
+            builder-eslint-config
+
+      - name: Publish package
+        uses: hostinger/actions/npm-publish@main
+        with:
+          bump_version: 'true'
```
Comment (IgnasCi) [inline .github/workflows/npm-publish-github-packages.yml:75]: bump false

**Pull request merged**